### PR TITLE
Add tags support

### LINKED ISSUE #24: To support tags
tags is not a supported feature yet.

### LINKED ISSUE #48: How to use tags within feature files and create @RunWith for it?
Hi,

I am trying to use tags like @smoke and @regression within features to create suites for my tests. Is there any way we can do this and run suites within runner class as we do with other cucumber tests?

### LINKED ISSUE #51: Skip scenarios
Does have any way to skip some scenarios by adding a tag `@skip` for example or another approach?

### PR DESCRIPTION
**Pull request by sawasawasawa**
##Overview This PR adds support for cucumber tag expressions, letting you run or skip tests, depending on their tags. 

The usage is based on cypress' env variable `TAGS`

Example: `cypress run -e TAGS="@smoke-tests and @ci"`

The main difference between this PR and #46 is the usage of cucumber's [tag expressions package](https://github.com/cucumber/cucumber/tree/master/tag-expressions), with support for its updated syntax. 
(`--tags ~@foo --tags @bar,@zap` becomes `--tags 'not @foo and (@bar or @zap)'`)

## Fixes

Fixes #51, fixes #48 and fixes #24.

## Credits

Many thanks to @ChrisBAshton and @GlynnJKW for their work in #46 and #58.

### CODE REVIEW

File: createTestsFromFeature.js [hunk 1]
```diff
@@ -1,17 +1,44 @@
...
-      createTestFromScenario(section, backgroundSection);
-    });
+  const featureTags = parsedFeature.feature.tags;
+  const hasEnvTags = !!getEnvTags();
+  const hasFeatureTags = featureTags && featureTags.length > 0;
+
+  let featureShouldRun = true;
+  if (hasEnvTags) {
+    if (hasFeatureTags) {
+      featureShouldRun = proceedCurrentStep(featureTags);
```
Comment (lgandecki) [inline createTestsFromFeature.js]: proceedCurrentStep name suggests an action, (proceeding the current step), but it returns a boolean. as with any booleans (function or variables) the name should read like part of a question sentence by starting with "is", "has", "should", and similar. 

if(proceedCurrentStep()) doesn't make sense

if (shouldProceedCurrentStep()) - even though a bit weird, makes all the sense in the world ;-)

**Comment by mikegfisher** [comment]
@lgandecki / @sawasawasawa do you know if this is going to get accepted and merged? 🤞

**Comment by mikegfisher** [comment]
> @lgandecki / @sawasawasawa do you know if this is going to get accepted and merged? 🤞

Nevermind, just saw [this](https://github.com/TheBrainFamily/cypress-cucumber-preprocessor/pull/58#issuecomment-426871265).

**Comment by sawasawasawa** [comment]
@mikegfisher hopefully soon :)

**Comment by sawasawasawa** [comment]
Fixes also #99 now.

File: createTestsFromFeature.js [hunk 2]
```diff
@@ -1,17 +1,47 @@
...
+  let featureShouldRun = true;
+  if (hasEnvTags) {
+    if (hasFeatureTags) {
+      featureShouldRun = shouldProceedCurrentStep(featureTags);
+    } else {
+      featureShouldRun = false;
+    }
+  }
+
+  const scenariosHaveTags = parsedFeature.feature.children.some(
```
Comment (lgandecki) [inline createTestsFromFeature.js]: this seems more like scenarioShouldRun? we need the scenario to have specific tags, not any tags

File: cypress/integration/TagsImplementation.feature [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    Given my cypress environment variable TAGS is '@foo and @bar'
+    Then tests tagged '@foo @bar' should proceed
+    And tests tagged '@foo @ba' should not proceed
+
+  Scenario: Passing multiple tags joined by OR operator
+    Given my cypress environment variable TAGS is '@foo or @bar'
+    Then tests tagged '@foo' should proceed
+    And tests tagged '@bar' should proceed
+    And tests tagged '@foo @baz' should proceed
+    And tests tagged '@fo @ba' should not proceed
```
Comment (lgandecki) [inline cypress/integration/TagsImplementation.feature]: 🥇

File: package.json [hunk 4]
```diff
@@ -4,9 +4,20 @@
   "description": "Run gherkin-syntaxed specs with cypress.io",
   "main": "index.js",
   "scripts": {
-    "test": "eslint . && jest && cypress run",
```
Comment (lgandecki) [inline package.json]: this way we don't lint on CI anymore. I think it's fine to have all 3 as tests, as per:

screen shot 2018-10-10 at 1 18 56 pm

**Comment by bbuckland** [comment]
@sawasawasawa @lgandecki Just wanted to follow up on this. My QA engineer has been begging for this feature :)

**Comment by lgandecki** [comment]
I believe this is ready.. Putting this on my todo list to release it tomorrow (it’s 9:30 pm where I currently am ;-) )

**Comment by lgandecki** [comment]
This doesn't fully work the way I would want it to, as it takes multiple seconds to check whether a particular file matches the requested tags. 
We should probably filter out the files BEFORE cypress "parses" them. (not sure how hard that would be though...)
But, it's a good start, and the syntax will stay the same.

**Comment by sabin-bhattarai-cko** [comment]
I am confused. Do @skip or @ignore tags work? Or if you could guide me on how could I have "cypress open" to follow certain tags. I want  " --tags ~@foo " in cypress UI.

**Review by vonhraban — COMMENTED** [review]
.

**Pull request merged**